### PR TITLE
[libc][bazel] Prevent LIBC_NAMESPACE leakage

### PR DESCRIPTION
**Pull request by gchatelet**
Right now `LIBC_NAMESPACE` definition is added to all rules that depend upon
any libc function. This have far-reaching effects which prevents llvm-libc
development when the installed libc is itself llvm-libc.

This PR removes the `:libc_root` target entirely and consistently use the
provided libc BUILD actions for compilation and testing. This makes sure that we
don't leak `LIBC_NAMESPACE` to our client.

### CODE REVIEW

**Comment by michaelrj-google** [comment]
I like the move to every target being through a `libc_*` instead of just `cc_*`. 
I'm not confident enough in my bazel to comment on the design of the rest of the change, but it works fine in my testing.

**Comment by gchatelet** [comment]
Thx for the review @michaelrj-google . Can you PTAL?

File: utils/bazel/llvm-project-overlay/libc/libc_build_rules.bzl [hunk 1]
```diff
@@ -48,7 +48,7 @@ def _libc_library(name, hidden, copts = [], deps = [], **kwargs):
 # Any library which does not define a public function should be listed with
 # libc_support_library.
 def libc_support_library(name, **kwargs):
-    _libc_library(name = name, hidden = True, **kwargs)
+    _libc_library(name = name, hidden = False, **kwargs)
```
Comment (lntue) [inline utils/bazel/llvm-project-overlay/libc/libc_build_rules.bzl:51]: Why are we not hiding support library symbols anymore?

**Review by michaelrj-google — APPROVED** [review]
Patch works on my end, please address Tue's comment before landing.

File: utils/bazel/llvm-project-overlay/libc/libc_build_rules.bzl (hunk 1 above)
Comment (gchatelet) [reply utils/bazel/llvm-project-overlay/libc/libc_build_rules.bzl:51]: Hiding the symbols results in internal fuzzers complaining about different visibility between modules:
```
default visibility for functions and variables [-fvisibility] differs in PCH file vs. current file
module file __redacted__/stl/_objs/stl_cc_library/stl_cc_library.pic.pcm cannot be loaded due to a configuration mismatch with the current compilation [-Wmodule-file-config-mismatch]
```

`hidden = False` is actually coherent with how it worked before this patch. I wanted to straighten up the visibility but it has consequences for downstream.

Now that I think about it, it makes sense that both internal targets and libc_support_library are compiled with the same visibility. I'll fix it by using `libc_support_library` in lieu of `_libc_library` in `libc_function` below.

Comment (gchatelet) [reply utils/bazel/llvm-project-overlay/libc/libc_build_rules.bzl:51]: @lntue does it look good to you?

**Review by lntue — APPROVED** [review]

**Pull request merged**